matrix-sdk/src/main/java/org/matrix/androidsdk/login/AutoDiscovery.kt

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.androidsdk.login
+
+import android.os.Handler
+import android.os.Looper
+import okhttp3.*
+import org.json.JSONObject
+import org.matrix.androidsdk.rest.callback.ApiCallback
+import org.matrix.androidsdk.util.Log
+import java.io.IOException
+import java.net.URL
+
+
+class AutoDiscovery(var okHttpClient: OkHttpClient = OkHttpClient()) {
+
+    companion object {
+        val instance = AutoDiscovery()
```

Comment:
If this class is a singleton, then it should be an object. But it's not. So I do not understand the aim of `instance`here.

Reply:
Was just utility, removed